File: src/main/java/ir/reservs/reservs/di/module/ApplicationModule.java
package ir.reservs.reservs.di.module;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {
    private final Application mApplication;

    public ApplicationModule(Application application) {
        mApplication = application;
    }

    @Provides
    @Singleton
    Context provideContext() {
        return mApplication.getApplicationContext();
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return mApplication;
    }


}

File: src/main/java/ir/reservs/reservs/di/module/ApplicationModule.kt
package ir.reservs.reservs.di.module

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ApplicationModule(private val mApplication: Application) {

    @Provides
    @Singleton
    fun provideContext(): Context {
        return mApplication.applicationContext
    }

    @Provides
    @Singleton
    fun provideApplication(): Application {
        return mApplication
    }

}
